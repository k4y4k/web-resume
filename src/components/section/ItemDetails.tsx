import * as React from 'react'
import tw, { css } from 'twin.macro'
import ReactMarkdown from 'react-markdown'

interface ItemDetailsTypes {
  // accepts markdown
  details?: string
}

const detailsStyles = css`
  ${tw`mt-2`}
  li {
    list-style-type: circle;
    list-style-position: inside;
  }
`

const ItemDetails = ({ details }: ItemDetailsTypes): JSX.Element | null => {
  if (details === '' || details === undefined) return null

  return (
    <div css={detailsStyles} data-testid='sectionItemDetails'>
      <ReactMarkdown children={details} />
    </div>
  )
}

export default ItemDetails
