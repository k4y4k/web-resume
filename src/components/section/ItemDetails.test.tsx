import * as React from 'react'
import { render, screen } from '@testing-library/react'
import ItemDetails from './ItemDetails'

describe('<ItemDetails />', () => {
  test('no data = no details section', () => {
    render(<ItemDetails details='' />)

    const details = screen.queryByTestId('sectionItemDetails')
    expect(details).not.toBeInTheDocument()
    expect(details).toMatchSnapshot()
  })

  test('renders markdown', () => {
    const string = '- a\n- b\n- c'

    render(<ItemDetails details={string} />)

    const details = screen.getByTestId('sectionItemDetails')
    expect(details).toBeInTheDocument()

    expect(details).toHaveTextContent('a b c')
    expect(details).toMatchSnapshot()
  })
})
