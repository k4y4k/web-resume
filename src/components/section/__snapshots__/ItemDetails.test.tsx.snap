// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemDetails /> renders markdown 1`] = `
<div
  css="
  ,margin-top: 0.5rem;,
  li {
    list-style-type: circle;
    list-style-position: inside;
  }
"
  data-testid="sectionItemDetails"
>
  <ul>
    <li>
      a
    </li>
    <li>
      b
    </li>
    <li>
      c
    </li>
  </ul>
</div>
`;
